### PR TITLE
docs: add example for awscc_cloudtrail_trail

### PR DESCRIPTION
**Pull request by BondAnthony**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request
* The resources and data sources in this provider are generated from the CloudFormation schema, so they can only support the actions that the underlying schema supports. For this reason submitted bugs should be limited to defects in the generation and runtime code of the provider. Customizing behavior of the resource, or noting a gap in behavior are not valid bugs and should be submitted as enhancements to AWS via the CloudFormation Open Coverage Roadmap.

Relates OR Closes #0000

### CODE REVIEW

**Review by quixoticmonk — APPROVED** [review]

File: examples/resources/awscc_cloudtrail_trail/complex_trail.tf [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    name = "Log all S3 objects events"
+    field_selectors = [
+      {
+        field  = "eventCategory"
+        equals = ["Data"]
+      },
+      {
+        field  = "resources.type"
+        equals = ["AWS::S3::Object"]
+      }
```
Comment (wellsiau-aws) [inline examples/resources/awscc_cloudtrail_trail/complex_trail.tf:20]:
```suggestion
      {
        field  = "resources.type"
        equals = ["AWS::S3::Object"]
      },
      {
        field  = "resources.ARN"
        equals = ["arn:aws:s3:::my-target-bucket/"]
      }
```

I suggest we scope this down to limit potential unintended cost overruns

**Review by wellsiau-aws — CHANGES_REQUESTED** [review]
Please have a look at the minor suggestion, otherwise looks good!

File: docs/resources/cloudtrail_trail.md [hunk 2]
```diff
@@ -10,6 +9,259 @@ description: |-
...
+  advanced_event_selectors = [{
+    name = "Log all S3 objects events"
+    field_selectors = [
+      {
+        field  = "eventCategory"
+        equals = ["Data"]
+      },
+      {
+        field  = "resources.type"
+        equals = ["AWS::S3::Object"]
```
[the remark this answers is not in the document]
Comment (BondAnthony) [reply docs/resources/cloudtrail_trail.md:104]: That's a good point. I'll provide an updated example.

[the remark this answers is not in the document]
Comment (BondAnthony) [reply docs/resources/cloudtrail_trail.md:104]: PR with the updated example. https://github.com/hashicorp/terraform-provider-awscc/pull/1619

**Pull request merged**